lib/bolt/transport/ssh/connection.rb

```diff
@@ -148,119 +142,86 @@ def disconnect
+                    "Could not execute command: #{command_str.inspect}",
+                    'EXEC_ERROR'
+                  )
+                end
 
-              channel.on_data do |_, data|
-                unless use_sudo && handled_sudo(channel, data, options[:stdin])
-                  result_output.stdout << data
+                channel.on_data do |_, data|
+                  out_wr << data
                 end
-                @logger.debug { "stdout: #{data.strip}" }
-              end
 
-              channel.on_extended_data do |_, _, data|
-                unless use_sudo && handled_sudo(channel, data, options[:stdin])
-                  result_output.stderr << data
+                channel.on_extended_data do |_, _, data|
+                  err_wr << data
                 end
-                @logger.debug { "stderr: #{data.strip}" }
-              end
 
-              channel.on_request("exit-status") do |_, data|
-                result_output.exit_code = data.read_long
+                channel.on_request("exit-status") do |_, data|
+                  exit_code = data.read_long
+                end
               end
-              # A wrapper is used to direct stdin when elevating privilage or using tty
-              if options[:stdin] && !use_sudo && !options[:wrapper]
-                channel.send_data(options[:stdin])
-                channel.eof!
+            end
+            write_th = Thread.new do
+              chunk_size = 4096
+              eof = false
+              active = true
+              readable = false
+              while active && !eof
+                @session.loop(0.1) do
+                  active = session_channel.active?
+                  readable = select([in_rd], [], [], 0)
+                  # Loop as long as the channel is still live and there's nothing to be written
+                  active && !readable
+                end
+                if readable
+                  if in_rd.eof?
+                    session_channel.eof!
+                    eof = true
+                  else
+                    to_write = in_rd.readpartial(chunk_size)
+                    session_channel.send_data(to_write)
+                  end
+                end
               end
+              session_channel.wait
             end
+            write_th.join
+            exit_code
+          ensure
+            write_th.terminate
+            in_rd.close
+            out_wr.close
+            err_wr.close
           end
-          session_channel.wait
+          [in_wr, out_rd, err_rd, th]
+        end
 
-          if result_output.exit_code == 0
-            @logger.debug { "Command returned successfully" }
-          else
-            @logger.info { "Command failed with exit code #{result_output.exit_code}" }
-          end
-          result_output
-        rescue StandardError
-          @logger.debug { "Command aborted" }
-          raise
+        def execute_simple(command)
```

Comment:
I don't see where this gets used.

Reply:
Whoops that was from earlier work, I'll remove it.